rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -250,28 +273,29 @@ def patch(self, deployment_id):
         request_dict = request.json
         sm = get_storage_manager()
         rm = get_resource_manager()
-        deployment = sm.get(models.Deployment, deployment_id)
-        allowed_attribs = {
-            'description', 'workflows', 'inputs', 'policy_types',
-            'policy_triggers', 'groups', 'scaling_groups', 'outputs',
-            'capabilities'
-        }
-        for attrib in allowed_attribs:
-            if attrib not in request_dict:
-                continue
-            previous = getattr(deployment, attrib, None)
-            if previous is not None:
-                raise ConflictError('{0} is already set'.format(attrib))
-            setattr(deployment, attrib, request_dict[attrib])
-        if 'labels' in request_dict:
-            raw_labels_list = request_dict.get('labels', [])
-            self._handle_deployment_labels(
-                sm,
-                rm,
-                deployment,
-                raw_labels_list
-            )
-        sm.update(deployment)
+        with sm.transaction():
+            deployment = sm.get(models.Deployment, deployment_id, locking=True)
+            allowed_attribs = {
+                'description', 'workflows', 'inputs', 'policy_types',
+                'policy_triggers', 'groups', 'scaling_groups', 'outputs',
+                'capabilities'
+            }
+            for attrib in allowed_attribs:
+                if attrib not in request_dict:
+                    continue
+                previous = getattr(deployment, attrib, None)
+                if previous is not None:
+                    raise ConflictError(f'{attrib} is already set')
+                setattr(deployment, attrib, request_dict[attrib])
+            if 'labels' in request_dict:
+                raw_labels_list = request_dict.get('labels', [])
+                self._handle_deployment_labels(
```

Comment:
@tehasdf I think here we could have have database deadlock since inside `_handle_deployment_labels` we will use locking for Deployment regarding counts + statuses.

Reply:
especially in case we have multiple deployments running at the same time

Reply:
i don't see the deadlock, this only locks the deployment

Reply:
do not forget we have another lock=True inside the statuses + counter propagation